### PR TITLE
apps/examples/wifi_manager_sample : Fix stress test1 crash during exit (ref: PR4888)

### PR DESCRIPTION
**Pull request by s1-vivek**
FileName: apps/examples/wifi_manager_sample/wm_test_stress_test1.c
Refactored stress 1 test case to handle error cases and do proper de-init.
This prevents error propagation from 1 iteration to another.

### CODE REVIEW

**Comment by s1-vivek** [comment]
> Hard code values should be replaced with #define or const
> Other changes are fine.

As discussed the changes are fine as per the coding guidelines.

**Review by deepaksrma — COMMENTED** [review]
could you paste the error logs in commit msg?
and why it was happening and how did you remove that?

**Comment by vibhor-m** [comment]
> could you paste the error logs in commit msg?
> and why it was happening and how did you remove that?

Commit amend done.

**Review by jeongchanKim — APPROVED** [review]
LGTM

File: apps/examples/wifi_manager_sample/wm_test_stress_test1.c [hunk 1]
```diff
@@ -122,20 +121,20 @@ static void wm_get_apinfo(wifi_manager_ap_config_s *apconfig)
...
 	ST_START_TEST;
+	ST_EXPECT_EQ(WIFI_MANAGER_SUCCESS, wifi_manager_deinit());
 	ST_END_TEST;
 }
 
 TEST_F(init)
 {
 	ST_START_TEST;
-	ST_EXPECT_EQ(WIFI_MANAGER_SUCCESS, wifi_manager_deinit());
+	ST_EXPECT_EQ(WIFI_MANAGER_SUCCESS, wifi_manager_init(&g_wifi_callbacks));
```
Comment (pillip8282) [inline apps/examples/wifi_manager_sample/wm_test_stress_test1.c:137]: the purpose of TEST(init) is check not only deinit result but also time duration of deinit. So deinit must be in TEST_F(init) function.

File: apps/examples/wifi_manager_sample/wm_test_stress_test1.c [hunk 2]
```diff
@@ -146,22 +145,22 @@ TEST_SETUP(connect)
...
 	ST_END_TEST;
 }
 
 TEST_F(connect)
 {
 	ST_START_TEST;
-	ST_EXPECT_EQ(WIFI_MANAGER_SUCCESS, wifi_manager_deinit());
+	wifi_manager_ap_config_s apconfig;
+	wm_get_apinfo(&apconfig);
+	ST_EXPECT_EQ(WIFI_MANAGER_SUCCESS, wifi_manager_connect_ap(&apconfig));
```
Comment (pillip8282) [inline apps/examples/wifi_manager_sample/wm_test_stress_test1.c:163]: same as above

**Review by pillip8282 — CHANGES_REQUESTED** [review]
I think you misunderstood the purpose of this test cases. deinit which is tested function must be inside TEST_F.

**Comment by s1-vivek** [comment]
> I think you misunderstood the purpose of this test cases. deinit which is tested function must be inside TEST_F.

will do changes and add deinit in TEST_F

File: apps/examples/wifi_manager_sample/wm_test_stress_test1.c (hunk 2 above)
Comment (s1-vivek) [reply apps/examples/wifi_manager_sample/wm_test_stress_test1.c:163]: purpose of connect, connect_done, disconnect and scan is to measure time of these respective operations.
Is it required to add deinit in TEST_F and add time of deinit execution in total TC execution time


> same as above

purpose of connect, connect_done, disconnect and scan is to measure time of these respective operations.
Is it required to add deinit in TEST_F and add time of deinit execution in total TC execution time

Comment (s1-vivek) [reply apps/examples/wifi_manager_sample/wm_test_stress_test1.c:163]: > same as above

purpose of connect, connect_done, disconnect and scan is to measure time of these respective operations.
Is it required to add deinit in TEST_F and add time of deinit execution in total TC execution time?

Comment (vibhor-m) [reply apps/examples/wifi_manager_sample/wm_test_stress_test1.c:163]: stress1 -> Connect_done ->
description: call deinit() after receiving connect done event

Originally code
TEST_SETUP -> init, connect ap,
TEST_F -> deinit
only deinit time was measured.

Modified code:
TEST_SETUP -> init,
TEST_F -> , connect ap
TEST_TEARDOWN -> deint
only connect_ap time is measured.

If required. we can add deinit in TEST_F.

**Comment by pillip8282** [comment]
> > I think you misunderstood the purpose of this test cases. deinit which is tested function must be inside TEST_F.
> 
> will do changes and add deinit in TEST_F

@s1-vivek 
Yes, the purpose of this TC is to measure deinit() accuracy and performance. So you have to add deinit in TEST_F

**Pull request closed without merging**